transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpResponseDecodingState.java

```diff
@@ -143,6 +143,9 @@ protected DecodingState finishDecode(IoBuffer readData, ProtocolDecoderOutput ou
                             httpResponse
                                     .setContent(new HttpContentMessage(allocator.wrap(allocator.allocate(0)), false));
                             out.write(httpResponse);
+                            if (status == HttpStatus.REDIRECT_NOT_MODIFIED) {
+                                return null;
+                            }
```

Comment:
RFC 7232 says "A 304 response cannot contain a message-body; it is always terminated by the first empty line after the header fields."

So, let us not do any content processing. I suppose something like the following should work :

```
            if (status == HttpStatus.REDIRECT_NOT_MODIFIED) {
                httpResponse.setContent(new HttpContentMessage(allocator.wrap(allocator.allocate(0)), true));
                out.write(httpResponse);
                return null;
            } else if (httpSession != null && httpSession.getMethod() == HttpMethod.HEAD) {
            ...
```